pipelines/scripts/update_spark_pool_references.py

```diff
@@ -0,0 +1,256 @@
+        self._spark_pool_cache = dict()
+        self._old_pool_name = old_pool_name
+        self._new_pool_name = new_pool_name
+    
+    def update_all_spark_pool_references(self):
+        """
+            Update all references to the old spark pool with the new spark pool
+        """
+        logging.info("Starting to replace spark pool references")
+        artifact_names = self.get_all_relevant_artifact_names()
+        logging.info("Performing replacement on the below files")
+        logging.info(json.dumps(artifact_names, indent=4))
+        logging.info("\nBegin replacement")
+        cleaned_artifacts = {
+            artifact_name: self.replace_spark_pool_references_in_artifact(artifact_name)
+            for artifact_name in artifact_names
+        }
+        logging.info("\nSaving changes\n")
+        for artifact_path, artifact_json in cleaned_artifacts.items():
+            logging.info(f"Overwriting artifact '{artifact_path}'")
+            with open(artifact_path, "w") as f:
+                json.dump(artifact_json, f, indent="\t", ensure_ascii=False)
+
+    def get_all_relevant_artifact_names(self) -> List[str]:
+        """
+            Return all local synapse artifact paths that should be updated
+        """
+        return [
+            os.path.join(path, name)
+            for path, subdirs, files in os.walk("workspace")
+            for name in files
+            if any(
+                os.path.join(path, name).startswith(x)
+                for x in [
+                    "workspace/notebook",
+                    "workspace/pipeline"
+                ]
+            )
+        ]
+
+    def get_spark_pool_details(self, spark_pool_name: str) -> Dict[str, Any]:
+        """
+            Get the details of the given spark pool
+
+            :param spark_pool_name: The name of the spark pool to get details for
+            :return: The details of the spark pool from the Synapse REST API
+        """
+        if spark_pool_name not in self._spark_pool_cache:
+            self._spark_pool_cache[spark_pool_name] = SynapseSparkPoolUtil(self._synapse_workspace_name).get(spark_pool_name)
+        return self._spark_pool_cache[spark_pool_name]
+
+    def replace_spark_pool_references_in_artifact(self, artifact_path: str) -> Dict[str, Any]:
+        """
+            Replace all references to the old spark pool with the new spark pool for the given artifact
+
+            :param artifact_path: The path of the artifact to update
+            :return: The artifact with updated references
+        """
+        if artifact_path.startswith("workspace/notebook"):
+            is_notebook = True
+        elif artifact_path.startswith("workspace/pipeline"):
+            is_notebook = False
+        else:
+            raise ValueError(f"Artifact '{artifact_path}' is not a path to a Synapse notebook or pipeline")
+        with open(artifact_path, "r") as f:
+            artifact = json.load(f)
```

Comment:
If a malformed or non-JSON file is present the script will crash here. Have you thought on adding a try/except around it ? 
Something like this : 
try:
    artifact = json.load(f)
except json.JSONDecodeError as e:
    logging.error(f"Failed to parse JSON in {artifact_path}: {e}")
    return {}

Reply:
I'll make a change to improve the logging

Reply:
I've made the change